Frontend/Gmfctn/src/app/modules/+auth/services/authenticate.service.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Observable, of, Subject } from 'rxjs';
+import { catchError, take, tap } from 'rxjs/operators';
+import { Tokens } from '../../../shared/models/token';
+import { apiUrl } from 'src/app/shared/environment/api-connection-string';
```

Comment:
Not custom env)
You have a folder `environment` inside `src` folder. There's two files `environment.ts` and `environment.prod.ts`.
Those files are there for the app global configurations, like apiUrl.
Please, move it there. 
If you'll need aditional information on this please reach me on scype.

Reply:
Fixed😎. 
I am not a dumb man, but I sometimes solve tasks in little weird way. 
It strikes me that tommorow I will have a lot of questions. 
Otake.